browser/ipfs/content_browser_client_helper_unittest.cc

```diff
@@ -137,4 +138,27 @@ TEST_F(ContentBrowserClientHelperUnitTest, HandleIPNSURLRewriteLocal) {
   ASSERT_TRUE(HandleIPFSURLRewrite(&ipns_uri, browser_context()));
 }
 
+TEST_F(ContentBrowserClientHelperUnitTest, HandleIPFSURLReverseRewriteLocal) {
+  profile()->GetPrefs()->SetInteger(
+      kIPFSResolveMethod, static_cast<int>(IPFSResolveMethodTypes::IPFS_LOCAL));
+  ASSERT_EQ(ipfs::GetConfiguredBaseGateway(browser_context(),
+    version_info::Channel::UNKNOWN),
+      GURL("http://localhost:48080/"));
+  GURL ipns_uri("http://test.com.ipns.localhost:8000");
+  ASSERT_FALSE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+  ipns_uri = GURL("http://test.com.ipns.localhost:48080");
+  ASSERT_TRUE(HandleIPFSURLReverseRewrite(&ipns_uri, browser_context()));
+}
+
+TEST_F(ContentBrowserClientHelperUnitTest, HandleIPFSURLReverseRewriteGateway) {
+  profile()->GetPrefs()->SetInteger(
+      kIPFSResolveMethod, static_cast<int>(IPFSResolveMethodTypes::IPFS_GATEWAY));
+  ASSERT_EQ(ipfs::GetConfiguredBaseGateway(
```

Comment:
We should also add a test case for getting the configured base gateway when the gateway is set to something like:
`http://localhost:8080`